### PR TITLE
[CI] Trigger elastic agent builds from the root pipeline

### PR DESCRIPTION
**Pull request by pazone**
## Proposed commit message
Trigger elastic agent builds from the root pipeline instead of `pull-requests.json`

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -581,3 +581,45 @@ steps:
...
+                trigger: "beats-xpack-elastic-agent"
+                build:
+                  commit: "${BUILDKITE_COMMIT}"
+                  branch: "${BUILDKITE_BRANCH}"
+                  env:
+                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
+                    - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+
+  - label: "Triggering Build for Packetbeat"
```
Comment (pchila) [inline .buildkite/pipeline.yml:620]:
```suggestion
  - label: "Triggering Build for Elastic Agent"
```

File: .buildkite/pull-requests.json [hunk 2]
```diff
@@ -15,22 +15,6 @@
...
-            "allowed_list": ["dependabot[bot]", "mergify[bot]"],
-            "set_commit_status": true,
-            "build_on_commit": true,
-            "build_on_comment": true,
-            "trigger_comment_regex": "^/test elastic-agent$",
-            "always_trigger_comment_regex": "^/test elastic-agent$",
-            "skip_ci_labels": [  ],
-            "skip_target_branches": [ ],
-            "skip_ci_on_only_changed": ["^x-pack/elastic-agent/README.md", "^x-pack/elastic-agent/docs/.*", "^x-pack/elastic-agent/devtools/.*" ],
-            "always_require_ci_on_changed": ["^x-pack/elastic-agent/.*", ".buildkite/x-pack/elastic-agent/.*", "^go.mod", "^pytest.ini", "^dev-tools/.*", "^libbeat/.*", "^testing/.*"]
```
Comment (dliappis) [inline .buildkite/pull-requests.json:33]: this is the same as in https://github.com/elastic/beats/pull/40026/files but in `7.17` branch right?

**Review by dliappis — APPROVED** [review]
LGTM

**Review by pchila — APPROVED** [review]

File: .buildkite/pull-requests.json (hunk 2 above)
Comment (pazone) [reply .buildkite/pull-requests.json:33]: yup. It's like reverse backport

**Pull request merged**